Public/Src/Sandbox/Windows/DetoursServices/FilesCheckedForAccess.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "FilesCheckedForAccess.h"
+#include "string.h"
+
+FilesCheckedForAccess FilesCheckedForAccess::Create()
+{
+    return FilesCheckedForAccess();
+}
+
+FilesCheckedForAccess::FilesCheckedForAccess()
+{
+    InitializeCriticalSection(&m_lock);
+}
+
+bool FilesCheckedForAccess::TryRegisterPath(const CanonicalizedPath& path) {
+    std::pair<std::unordered_set<std::wstring>::iterator, bool> result;
+    
+    EnterCriticalSection(&m_lock);
+    result = m_pathSet.insert(path.GetPathString());
+    LeaveCriticalSection(&m_lock);
+
+    return result.second;
+}
+
+bool FilesCheckedForAccess::IsRegistered(const CanonicalizedPath& path) {

```

Comment:
Where is this used?

Reply:
Nowhere as of now. Just added it for completeness.

---
In reply to: [299198574](https://github.com/microsoft/BuildXL/pull/536#discussion_r299198574) [](ancestors = 299198574)